opennms-base-assembly/src/main/filtered/bin/find-java.sh

```diff
@@ -1,7 +1,7 @@
-#!/bin/sh
+#!/bin/bash
 
-JAVA_SEARCH_DIRS="$JAVA_SEARCH_DIRS /usr/lib/jvm /usr/java /System/Library/Java/JavaVirtualMachines /Library/Java/JavaVirtualMachines /Library/Java/Home /opt"
-# set DEBUG=true to enable debugging
+# if $JAVA_SEARCH_DIRS is already set, make sure it is treated as an array
+declare -p JAVA_SEARCH_DIRS 2>/dev/null | grep -q '^declare \-a' || IFS=" " read -r -a JAVA_SEARCH_DIRS <<< "${JAVA_SEARCH_DIRS[@]}"
```

Comment:
Would this do the same thing? This is implicitly treating it as an array and then storing that in an array variable.

```suggestion
JAVA_SEARCH_DIRS=(${JAVA_SEARCH_DIRS[@]})
```

```
OpenNMSs-MacBook-Pro:Downloads opennms$ testString="hello world"
OpenNMSs-MacBook-Pro:Downloads opennms$ declare -p testString
declare -- testString="hello world"
OpenNMSs-MacBook-Pro:Downloads opennms$ testString=(${testString[@]})
OpenNMSs-MacBook-Pro:Downloads opennms$ declare -p testString
declare -a testString='([0]="hello" [1]="world")'
OpenNMSs-MacBook-Pro:Downloads opennms$ declare -p doesNotExist
-bash: declare: doesNotExist: not found
OpenNMSs-MacBook-Pro:Downloads opennms$ test2=(${doesNotExist})
OpenNMSs-MacBook-Pro:Downloads opennms$ declare -p test2
declare -a test2='()'
```

Reply:
turns out it gets even more complicated than this, new code handles it